tests/CMakeLists.txt

```diff
@@ -0,0 +1,106 @@
+cmake_minimum_required(VERSION 3.5)
+project(unittest)
+
+set(EXAMPLES "matmul;minimal" CACHE STRING "a list of examples to run as tests")
+set(SAMPLES hosvd matmul minimal seissol_eqspp stock tce)
+set_property(CACHE EXAMPLES PROPERTY STRINGS ${SAMPLES})
+
+set(VARIANT "Eigen" CACHE STRING "example-specific variant")
+set(VARIANTS Eigen LIBXSMM OpenBLAS)
+set_property(CACHE VARIANT PROPERTY STRINGS ${VARIANTS})
+
+set(PRECISION "double" CACHE STRING "floating point type: double/single")
+set(PRECISION_OPTIONS "double" "single")
+set_property(CACHE PRECISION PROPERTY STRINGS ${PRECISION_OPTIONS})
+
+set(ARCH "hsw" CACHE STRING "floating point type: double/single")
+set(ARCH_OPTIONS snb hsw skx thunderx2t99 nvidia)
+set_property(CACHE ARCH PROPERTY STRINGS ${ARCH_OPTIONS})
+
+# adjust cli arg. for the code gen.
+if (${PRECISION} STREQUAL "double")
+    set(ARCH_STR d${ARCH})
+else()
+    set(ARCH_STR s${ARCH})
+endif()
+
+# find appropriate compiler flags
+if (${ARCH} STREQUAL "snb")
+    set(CPU_FLAGS "-mavx")
+elseif (${ARCH} STREQUAL "hsw")
+    set(CPU_FLAGS "-march=core-avx2")
+elseif(${ARCH} STREQUAL "skx")
+    set(CPU_FLAGS "-xCORE-AVX512" "-fma")
```

Comment:
These do not look optimal - couldn't we just use -mcpu=native -mtune=native (one of both is good enough, can't remember which one imples which) for unit tests as default?

Reply:
I just took it from SeisSol/cmake/cpu_arch_flags.cmake. Let's make just
```
-mcpu=native
```
for all architectures. I guess a compiler should figure out which flags to pick up

Reply:
I think this should be -mcpu=native for arm and -march=native for all other architectures. Please don't ask why, but -mcpu is a deprecated synonym for -mtune of x86 but this is not the case for ARM. And mcpu doesn't give the best performance, you'd need to use -march then.

> In short, you cannot support both x86 and Arm with exactly the same compiler flags.  You must use -mcpu for Arm and -march for x86.

See [here](https://community.arm.com/developer/tools-software/tools/b/tools-software-ides-blog/posts/compiler-flags-across-architectures-march-mtune-and-mcpu) for all the details you didn't want to know.

Sorry for being pedantic about this - the reason I am is that I also wanted to have =native settings for SeisSol. Still not sure what flags we'd need to set...